File: app/src/main/java/com/app/share/fragments/NewsFragment.java
package com.app.share.fragments;

import android.graphics.drawable.Drawable;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.graphics.drawable.DrawableCompat;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.app.share.R;
import com.app.share.fragments.newsfragments.CryptoFragment;
import com.app.share.fragments.newsfragments.GlobalFragment;
import com.app.share.fragments.newsfragments.IPOFragment;
import com.app.share.fragments.newsfragments.MarketFragment;
import com.app.share.fragments.newsfragments.StocksFragment;

public class NewsFragment extends Fragment {
    TextView markets, stocks, crypto, IPO, global;
    View view;
    private String webURL = "https://m.economictimes.com/";

    //NewsFragments
    MarketFragment marketFragment;
    StocksFragment stocksFragment;
    CryptoFragment cryptoFragment;
    IPOFragment ipoFragment;
    GlobalFragment globalFragment;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.fragment_news, container, false);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        initViews();
        intializeNewsFragments();

        FragmentTransaction ft2 = getChildFragmentManager().beginTransaction();
        ft2.add(R.id.framelayout, marketFragment);
        ft2.add(R.id.framelayout, stocksFragment);
        ft2.add(R.id.framelayout, cryptoFragment);
        ft2.add(R.id.framelayout, ipoFragment);
        ft2.add(R.id.framelayout, globalFragment);

        ft2.hide(stocksFragment);
        ft2.hide(cryptoFragment);
        ft2.hide(ipoFragment);
        ft2.hide(globalFragment);
        ft2.commit();

        performActions();

    }

    private void replaceInnerFragment(Fragment fragment) {
        FragmentTransaction ft2 = getChildFragmentManager().beginTransaction();
        ft2.hide(marketFragment);
        ft2.hide(stocksFragment);
        ft2.hide(cryptoFragment);
        ft2.hide(ipoFragment);
        ft2.hide(globalFragment);

        ft2.show(fragment);
        ft2.commit();
    }

    private void intializeNewsFragments() {
        marketFragment = new MarketFragment();
        stocksFragment = new StocksFragment();
        cryptoFragment = new CryptoFragment();
        ipoFragment = new IPOFragment();
        globalFragment = new GlobalFragment();
    }

    private void performActions() {
        Drawable drawablebtn = markets.getBackground();
        drawablebtn = DrawableCompat.wrap(drawablebtn);
        DrawableCompat.setTint(drawablebtn, getResources().getColor(R.color.black));
        markets.setBackground(drawablebtn);

        markets.setOnClickListener(view -> {
            Drawable buttonDrawable = markets.getBackground();
            buttonDrawable = DrawableCompat.wrap(buttonDrawable);
            DrawableCompat.setTint(buttonDrawable, getResources().getColor(R.color.green));
            replaceInnerFragment(marketFragment);

            Drawable stockBtn = stocks.getBackground();
            stockBtn = DrawableCompat.wrap(stockBtn);
            DrawableCompat.setTint(stockBtn, getResources().getColor(R.color.light_gray));
            stocks.setBackground(stockBtn);

            Drawable cryptoBtn = crypto.getBackground();
            cryptoBtn = DrawableCompat.wrap(cryptoBtn);
            DrawableCompat.setTint(cryptoBtn, getResources().getColor(R.color.light_gray));
            crypto.setBackground(cryptoBtn);

            Drawable ipoBtn = IPO.getBackground();
            ipoBtn = DrawableCompat.wrap(ipoBtn);
            DrawableCompat.setTint(ipoBtn, getResources().getColor(R.color.light_gray));
            IPO.setBackground(ipoBtn);

            Drawable globalBtn = global.getBackground();
            globalBtn = DrawableCompat.wrap(globalBtn);
            DrawableCompat.setTint(globalBtn, getResources().getColor(R.color.light_gray));
            global.setBackground(globalBtn);
        });

        stocks.setOnClickListener(view -> {
            Drawable buttonDrawable = stocks.getBackground();
            buttonDrawable = DrawableCompat.wrap(buttonDrawable);
            DrawableCompat.setTint(buttonDrawable, getResources().getColor(R.color.green));
            stocks.setBackground(buttonDrawable);
            replaceInnerFragment(stocksFragment);

            Drawable marketBtn = markets.getBackground();
            marketBtn = DrawableCompat.wrap(marketBtn);
            DrawableCompat.setTint(marketBtn, getResources().getColor(R.color.light_gray));
            markets.setBackground(marketBtn);

            Drawable cryptoBtn = crypto.getBackground();
            cryptoBtn = DrawableCompat.wrap(cryptoBtn);
            DrawableCompat.setTint(cryptoBtn, getResources().getColor(R.color.light_gray));
            crypto.setBackground(cryptoBtn);

            Drawable ipoBtn = IPO.getBackground();
            ipoBtn = DrawableCompat.wrap(ipoBtn);
            DrawableCompat.setTint(ipoBtn, getResources().getColor(R.color.light_gray));
            IPO.setBackground(ipoBtn);

            Drawable globalBtn = global.getBackground();
            globalBtn = DrawableCompat.wrap(globalBtn);
            DrawableCompat.setTint(globalBtn, getResources().getColor(R.color.light_gray));
            global.setBackground(globalBtn);
        });

        crypto.setOnClickListener(view -> {
            Drawable buttonDrawable = crypto.getBackground();
            buttonDrawable = DrawableCompat.wrap(buttonDrawable);
            DrawableCompat.setTint(buttonDrawable, getResources().getColor(R.color.green));
            crypto.setBackground(buttonDrawable);
            replaceInnerFragment(cryptoFragment);

            Drawable marketBtn = markets.getBackground();
            marketBtn = DrawableCompat.wrap(marketBtn);
            DrawableCompat.setTint(marketBtn, getResources().getColor(R.color.light_gray));
            markets.setBackground(marketBtn);

            Drawable stockBtn = stocks.getBackground();
            stockBtn = DrawableCompat.wrap(stockBtn);
            DrawableCompat.setTint(stockBtn, getResources().getColor(R.color.light_gray));
            stocks.setBackground(stockBtn);

            Drawable ipoBtn = IPO.getBackground();
            ipoBtn = DrawableCompat.wrap(ipoBtn);
            DrawableCompat.setTint(ipoBtn, getResources().getColor(R.color.light_gray));
            IPO.setBackground(ipoBtn);

            Drawable globalBtn = global.getBackground();
            globalBtn = DrawableCompat.wrap(globalBtn);
            DrawableCompat.setTint(globalBtn, getResources().getColor(R.color.light_gray));
            global.setBackground(globalBtn);

        });

        IPO.setOnClickListener(view -> {
            Drawable buttonDrawable = IPO.getBackground();
            buttonDrawable = DrawableCompat.wrap(buttonDrawable);
            DrawableCompat.setTint(buttonDrawable, getResources().getColor(R.color.green));
            IPO.setBackground(buttonDrawable);
            replaceInnerFragment(ipoFragment);

            Drawable marketBtn = markets.getBackground();
            marketBtn = DrawableCompat.wrap(marketBtn);
            DrawableCompat.setTint(marketBtn, getResources().getColor(R.color.light_gray));
            markets.setBackground(marketBtn);

            Drawable stockBtn = stocks.getBackground();
            stockBtn = DrawableCompat.wrap(stockBtn);
            DrawableCompat.setTint(stockBtn, getResources().getColor(R.color.light_gray));
            stocks.setBackground(stockBtn);

            Drawable cryptoBtn = crypto.getBackground();
            cryptoBtn = DrawableCompat.wrap(cryptoBtn);
            DrawableCompat.setTint(cryptoBtn, getResources().getColor(R.color.light_gray));
            crypto.setBackground(cryptoBtn);

            Drawable globalBtn = global.getBackground();
            globalBtn = DrawableCompat.wrap(globalBtn);
            DrawableCompat.setTint(globalBtn, getResources().getColor(R.color.light_gray));
            global.setBackground(globalBtn);


        });

        global.setOnClickListener(view -> {
            Drawable buttonDrawable = global.getBackground();
            buttonDrawable = DrawableCompat.wrap(buttonDrawable);
            DrawableCompat.setTint(buttonDrawable, getResources().getColor(R.color.green));
            global.setBackground(buttonDrawable);
            replaceInnerFragment(globalFragment);

            Drawable marketBtn = markets.getBackground();
            marketBtn = DrawableCompat.wrap(marketBtn);
            DrawableCompat.setTint(marketBtn, getResources().getColor(R.color.light_gray));
            markets.setBackground(marketBtn);

            Drawable stockBtn = stocks.getBackground();
            stockBtn = DrawableCompat.wrap(stockBtn);
            DrawableCompat.setTint(stockBtn, getResources().getColor(R.color.light_gray));
            stocks.setBackground(stockBtn);

            Drawable cryptoBtn = crypto.getBackground();
            cryptoBtn = DrawableCompat.wrap(cryptoBtn);
            DrawableCompat.setTint(cryptoBtn, getResources().getColor(R.color.light_gray));
            crypto.setBackground(cryptoBtn);

            Drawable ipoBtn = IPO.getBackground();
            ipoBtn = DrawableCompat.wrap(ipoBtn);
            DrawableCompat.setTint(ipoBtn, getResources().getColor(R.color.light_gray));
            IPO.setBackground(ipoBtn);

        });
    }

    private void initViews() {
        markets = view.findViewById(R.id.tvMarkets);
        stocks = view.findViewById(R.id.tvStocks);
        crypto = view.findViewById(R.id.tvCrypto);
        IPO = view.findViewById(R.id.tvIPO);
        global = view.findViewById(R.id.tvGlobal);
    }

    @Override
    public void onResume() {
        super.onResume();
    }

}

File: app/src/main/java/com/app/share/fragments/NewsFragment.kt
package com.app.share.fragments

import android.annotation.SuppressLint
import android.content.Intent
import android.graphics.Bitmap
import android.widget.TextView
import com.app.share.fragments.newsfragments.MarketFragment
import com.app.share.fragments.newsfragments.StocksFragment
import com.app.share.fragments.newsfragments.CryptoFragment
import com.app.share.fragments.newsfragments.IPOFragment
import com.app.share.fragments.newsfragments.GlobalFragment
import android.os.Bundle
import com.app.share.R
import android.graphics.drawable.Drawable
import android.view.*
import android.webkit.*
import android.widget.ImageView
import android.widget.ProgressBar
import androidx.core.graphics.drawable.DrawableCompat
import androidx.fragment.app.Fragment
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import com.app.share.activities.HelpActivity

class NewsFragment : Fragment() {
    var webView: WebView? = null
    var webURL: String? = "https://www.amazon.in/"
    var swipeRefreshLayout: SwipeRefreshLayout? = null
    var progressBar: ProgressBar? = null
    var ivBack: ImageView? = null

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_news, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        webView = view.findViewById(R.id.webview)
        swipeRefreshLayout = view.findViewById(R.id.swipe)
        swipeRefreshLayout?.setOnRefreshListener {
            webURL = webView?.url
            refreshWebView()
            swipeRefreshLayout?.isRefreshing = false
        }
        progressBar = view.findViewById(R.id.seekbar)
        webView = view.findViewById(R.id.webview)
        ivBack = view.findViewById(R.id.ivBack)
        ivBack?.setOnClickListener {
            if (webView?.canGoBack()!!) {
                webView?.goBack()
            } else {
                requireActivity().onBackPressed()
            }
        }

        setWebView()
    }

    private fun refreshWebView() {
        webView!!.webViewClient = webViewClient
        webView!!.loadUrl(webURL!!)
    }

    @SuppressLint("SetJavaScriptEnabled")
    private fun setWebView() {
        webView?.webViewClient = webViewClient
        webView!!.settings.setRenderPriority(WebSettings.RenderPriority.HIGH)
        webView!!.settings.javaScriptEnabled = true
        webView!!.setLayerType(View.LAYER_TYPE_HARDWARE, null)
        webView!!.settings.cacheMode = WebSettings.LOAD_NO_CACHE
        webView!!.settings.setAppCacheEnabled(true)
        webView!!.settings.domStorageEnabled = true
        webView!!.overScrollMode = WebView.OVER_SCROLL_NEVER
        CookieManager.getInstance().setAcceptThirdPartyCookies(webView, true)

        webView!!.webChromeClient = webChromeClient

        webView!!.setOnKeyListener { v: View?, keyCode: Int, event: KeyEvent ->
            if (keyCode == KeyEvent.KEYCODE_BACK && event.action == MotionEvent.ACTION_UP && webView!!.canGoBack()) {
                webView!!.goBack()
                return@setOnKeyListener true
            }
            false
        }

        webView!!.loadUrl(webURL.toString())
    }

    val webChromeClient = object : WebChromeClient() {
        override fun onProgressChanged(view: WebView, progress: Int) {
            progressBar!!.progress = progress
            if (progress == 100) progressBar!!.visibility = View.GONE
        }
    }
    val webViewClient = object : WebViewClient() {
        override fun shouldOverrideUrlLoading(view: WebView?, url: String?): Boolean {
            view?.loadUrl(url.toString())
            return true
        }

        override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) {
            super.onPageStarted(view, url, favicon)
            progressBar!!.visibility = View.VISIBLE
        }

        override fun onPageFinished(view: WebView?, url: String?) {
            super.onPageFinished(view, url)
            progressBar!!.visibility = View.GONE
        }
    }
}